language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/Contract.java

```diff
@@ -48,9 +48,6 @@ protected Contract(
     this.observers = observers;
   }
 
-  // concrete 1st type param would need a self-reference type param in Contract
-  protected abstract ContractCompanion<? extends Contract<Id, Data>, Id, Data> getCompanion();
```

Comment:
You should be able to change the type to `ContractTypeCompanion<?, Data>` and keep the name, since you have such a thing in the interface case.